src/index.js

```diff
@@ -55,20 +78,38 @@ module.exports = declare(api => {
               }
             })
 
+            // Detect if parent is React component or DOM node
+            // Option adds attrs to first DOM node in the component
+            // and ignores inner nodes
+            const firstChild = firstChildOnly
```

Comment:
Could we rename these two vars (`firstChild` and `filter`) to something else? They seem to mean "does this element conform to the firstChildOnly rule" and "does this child matches filter, if defined". 
When I initially saw `firstChild`, especially given line 105, it seems to suggest "is this element the first child", and got a bit confused.
I'd suggest something like `matchFirstChildRule` or `matchFirstChildCondition` (I'm horrible at naming so feel free to change to other stuff).